File: LeetCode/[M] - 328. Odd Even Linked List/solution.go
package main

// ListNode - Definition for singly-linked list.
type ListNode struct {
	Val  int
	Next *ListNode
}

func oddEvenList(head *ListNode) *ListNode {
	if head == nil || head.Next == nil || head.Next.Next == nil {
		return head
	}
	headEven := head.Next
	tailOdd, tailEven := head, headEven
	for tailEven != nil && tailEven.Next != nil {
		tailOdd.Next = tailOdd.Next.Next
		tailEven.Next = tailEven.Next.Next
		tailOdd = tailOdd.Next
		tailEven = tailEven.Next
	}
	tailOdd.Next = headEven
	return head
}

File: LeetCode/[M] - 328. Odd Even Linked List/solution.py
from typing import Optional


# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def oddEvenList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head or not head.next or not head.next.next:
            return head
        headEven = head.next
        tailOdd, tailEven = head, headEven
        while tailEven and tailEven.next:
            tailOdd.next = tailOdd.next.next
            tailEven.next = tailEven.next.next
            tailOdd = tailOdd.next
            tailEven = tailEven.next
        tailOdd.next = headEven
        return head
